desc/objectives/utils.py

```diff
@@ -130,32 +132,49 @@ def factorize_linear_constraints(objective, constraint):  # noqa: C901
             )
         A = A[unfixed_rows][:, unfixed_idx]
         b = b[unfixed_rows]
+
+    # unscaled particular solution to get scale of x
```

Comment:
This seems a little wasteful because we are now performing the SVD twice: once for the original system to get the scale of the variables, then again for the scaled system. 

The issue is we need a way to know the scale of the full state vector, and the particular solution `xp` gives us that. Otherwise we would have add the optimizable `things` as an input to this function so that we can compute the initial state vector `x0` and use that as the scale. 

In practice I don't think this SVD takes much computation time so I'm not too worried about it. But I'm open to suggestions for improvement here.

Reply:
To get `xp` we don't need the null space, so in theory we don't need to use an SVD at all, we could do a regular QR least norm which would in theory be quite a bit faster.

That said, I think basing the scale of x0 might have some other benefits, such as properly rescaling variables that aren't necessarily being constrained. IE, for bootstrap current optimization the current profile is usually free so it wouldn't show up in any particular solution and hence wouldn't get scaled, though it may help the optimizer if it was.